src/quit_confirmation.cpp

```diff
@@ -59,5 +62,26 @@ bool quit_confirmation::show_prompt(const std::string& message)
 
 bool quit_confirmation::default_prompt()
 {
-	return show_prompt(_("Do you really want to quit?"));
+	playmp_controller* pmc = dynamic_cast<playmp_controller*>(resources::controller);
```

Comment:
It seems to me the only reason you need a playmp_controller is for the surrender() call. I suggest moving the surrender() function to play_controller. I see no reason to limit this to mp even if it might not make too much sense in sp.

Then you can get rid of the ugly dynamic_cast.